DDD-1.md

```diff
@@ -0,0 +1,173 @@
+2. Cluster stability. If the resources above are insufficient (e.g. memory), it may cause a Connect worker crash which in turn will fail all connector tasks scheduled to the worker. In the best case scenario, those tasks can be just restarted. In a worse case scenario, the restarted connectors will produce duplicates since the source connectors currently don't implement exactly-once processing semantics. In an even worse case, a worker can fail during a long-running snapshot, handling which will require additional cleanup.
+3. SQL Server resources. Each server instance will have to maintain an extra set of persistent connections (one per database, tens to hundreds in total) which may require additional server resources.
+
+## Proposed Changes
+
+### Source partition awareness at the task level ([#15](https://github.com/sugarcrm/debezium/pull/15))
+
+Currently, a Debezium connector task can process a single source partition throught its lifecycle. This partition is usually hard-coded in the context initialization code. E.g.:
+
+```java
+public SqlServerOffsetContext(SqlServerConnectorConfig connectorConfig) {
+        partition = Collections.singletonMap(SERVER_PARTITION_KEY, connectorConfig.getLogicalName());
+```
+
+In order to achieve partition-awareness, we need to introduce an API and a set of components that do the following:
+
+1. Describe a *set* of source partitions to be processed by the task.
+2. Initialize the set above. Normally, based on the task configuration.
+3. Represent a source partition in higher-level terms specific to a given connector (e.g. database name).
+4. Allow mapping of connector-level partition (e.g. a database name) to the corresponding Kafka Connect source partition.
+5. Allow mapping of connector-level partition to the corresponding `OffsetContext`.
+6. Instead of a single `OffsetContext` per task, use a map of connector-level partitions to their offset context.
+
+### Support for multiple tasks per connector ([#9](https://github.com/sugarcrm/debezium/issues/9))
+
+Apart from being able to process multiple databases by a given task, it would be great if a connector was able to start multiple tasks and divide the databases between them. This way, it would allow to scale a single connector just by changing its `max.tasks` configuration.
+
+Being able to specify which subset of the databases a given task should process warrants the introduction of a new task-level configuration parameter ([#17](https://github.com/sugarcrm/debezium/issues/17)):
+
+| Name                  | Type     | Default | Importance | Doc                                                          |
+| :-------------------- | :------- | :------ | :--------- | :----------------------------------------------------------- |
+| `task.database.names` | `STRING` | `None`  | `HIGH`     | The list of comma-separated database names form which the given connector task should capture the changes from. Must be non-empty. |
```

Comment:
Is this something the user needs to provide? Or could the connector itself organize that mapping from database(s) to task(s)?

Reply:
It's the connector that will read its `database.names` and divide them between tasks (`task.database.names`). This configuration will be internal to the connector.

Reply:
Perfect.